derive_action_signing/module_data/deposit.py

```diff
@@ -15,11 +18,20 @@ class DepositModuleData(ModuleData):
     decimals: int
 
     def to_abi_encoded(self):
+        amount_decimal = Decimal(self.amount)
+        scaled_amount = int(amount_decimal.scaleb(self.decimals))
         return encode(
-            ["uint", "address", "address"],
+            ["uint256", "address", "address"],
             [
-                int(self.amount * Decimal(10) ** self.decimals),
+                scaled_amount,
                 Web3.to_checksum_address(self.asset),
                 Web3.to_checksum_address(self.manager),
             ],
         )
+
+    def to_json(self):
+        return {
+            "amount": str(self.amount),
+            "asset_name": "USDC",
```

Comment:
TODO: ensure not just usdc

Reply:
this can be any asset, not just USDC so should return `self.asset_name`